validator/specs/lib/validateB3dmSpec.js

```diff
@@ -33,37 +53,140 @@ function createB3dmTile() {
 }
 
 function createInvalidMagic() {
-    var header = new Buffer(24);
+    var header = createB3dmTile();
     header.write('xxxx', 0); // magic
-    header.writeUInt32LE(1, 4); // version
-    header.writeUInt32LE(header.length, 8); // byteLength
-    header.writeUInt32LE(0, 12); // batchTableJSONByteLength
-    header.writeUInt32LE(0, 16); // batchTableBinaryByteLength
-    header.writeUInt32LE(0, 20); // batchLength
-
+    console.log('checking:\n' + header + '\n');
     return header;
 }
 
 function createInvalidVersion() {
-    var header = new Buffer(24);
-    header.write('b3dm', 0); // magic
+    var header = createB3dmTile();
     header.writeUInt32LE(5, 4); // version
-    header.writeUInt32LE(header.length, 8); // byteLength
-    header.writeUInt32LE(0, 12); // batchTableJSONByteLength
-    header.writeUInt32LE(0, 16); // batchTableBinaryByteLength
-    header.writeUInt32LE(0, 20); // batchLength
 
     return header;
 }
 
 function createWrongByteLength() {
-    var header = new Buffer(24);
-    header.write('b3dm', 0); // magic
-    header.writeUInt32LE(1, 4); // version
+    var header = createB3dmTile();
     header.writeUInt32LE(header.length - 1, 8); // byteLength
-    header.writeUInt32LE(0, 12); // batchTableJSONByteLength
-    header.writeUInt32LE(0, 16); // batchTableBinaryByteLength
-    header.writeUInt32LE(0, 20); // batchLength
 
     return header;
+}
+
+function createB3dmBatchJson() {
+    var header = createB3dmTile();
+    var batchJSON = createValidBatchTableJSON();
+    header.writeUInt32LE(header.length + batchJSON.length, 8); // byteLength
+    header.writeUInt32LE(batchJSON.length, 12); // batchTableJSONByteLength
+
+    return Buffer.concat([header, batchJSON]);
```

Comment:
Better to avoid condensing the name, `batchTableJSON` while longer is easier to follow.